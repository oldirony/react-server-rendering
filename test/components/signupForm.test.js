import React from 'react';
import Signup from '../../src/components/signup-form/signup-form';

const props = {
	handleSubmit: func => func.bind(null, {})
}

test('It renders correctly', ()=>{
	let form = shallow(<Signup {...props}/>);
	expect(form).toMatchSnapshot();
})

test('It renders error message correctly', ()=>{
	let form = shallow(<Signup {...props} authError="Error message"/>);
	expect(form).toMatchSnapshot();
})