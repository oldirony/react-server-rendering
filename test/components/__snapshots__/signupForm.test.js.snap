// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Sign up now
  </h1>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <Field
        className="input"
        component={[Function]}
        name="email"
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="row"
    >
      <Field
        className="input"
        component={[Function]}
        name="password"
        placeholder="Password"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div
      className="row"
    >
      <Field
        className="input"
        component={[Function]}
        name="repeatPassword"
        placeholder="Repeat password"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <button
        className="button fullwidth"
        type="submit"
      >
        Sign up
      </button>
    </div>
    <Link
      className="vanilla"
      replace={false}
      to="/login"
    >
      Do you already have an account?
    </Link>
  </div>
</form>
`;

exports[`It renders error message correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="authError"
  >
    Error message
  </div>
  <h1>
    Sign up now
  </h1>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <Field
        className="input"
        component={[Function]}
        name="email"
        placeholder="Email"
        type="email"
      />
    </div>
    <div
      className="row"
    >
      <Field
        className="input"
        component={[Function]}
        name="password"
        placeholder="Password"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div
      className="row"
    >
      <Field
        className="input"
        component={[Function]}
        name="repeatPassword"
        placeholder="Repeat password"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
  </div>
  <div
    className="section"
  >
    <div
      className="row"
    >
      <button
        className="button fullwidth"
        type="submit"
      >
        Sign up
      </button>
    </div>
    <Link
      className="vanilla"
      replace={false}
      to="/login"
    >
      Do you already have an account?
    </Link>
  </div>
</form>
`;
